### PR TITLE
Refactor/HDF-95 Refactor Rewards worker to import halo-rewards contracts via git submodule

### PR DESCRIPTION
**Pull request by mystbrent**
## Description of Changes

Before this update, the repository explicitly includes halo-reward contracts to the source code to instantiate the Rewards contract. This can be improved by using git submodules to easily reference the halo-rewards contracts to its github repository counterpart.

Added changes also include:

[x] Add postinstall npm script to `package.json` to auto install `halo-rewards` submodule deps
[x] Make use of typescript path module aliases when importing halo-rewards repo to source code

With regards to deriving the signer account from only mnemonic option, I think it limits the option for the worker runner to only one option of using mnemonic. This update also includes resolving that issue by prioritizing the requirement of private key first before mnemonic. This way local tests can easily be done without having to do much effort.

[Link to Jira Ticket](https://halodao.atlassian.net/browse/HDF-95)

### Developer Checklist:

- [x] I have followed the guidelines in our Contributing document
- [x] This PR has a corresponding JIRA ticket
- [x] My branch conforms with our naming convention i.e. `feature/HDF-XXX-description`
- [x] I have written new tests for your core changes, as applicable
- [x] I have successfully ran tests locally
- [x] I have formatted my code using format document in VSCode

### Reviewers Checklist:

- [ ] Code is readable and understandable; any unclear parts have explanations
- [ ] UI/UX changes match the corresponding figma/other design resources, if applicable
- [ ] I have successfully ran tests locally

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[submodule "halo-rewards"]
+	path = halo-rewards
+	url = git@github.com:HaloDAO/halo-rewards.git
```
Comment (tracyarciaga) [inline .gitmodules:3]: Please change url = https://github.com/HaloDAO/halo-rewards.git so that it uses https instead of ssh.

**Review by tracyarciaga — CHANGES_REQUESTED** [review]

File: .gitmodules (hunk 1 above)
Comment (bitcoinbrisbane) [reply .gitmodules:3]: I use `SSH`

**Review by bitcoinbrisbane — APPROVED** [review]

**Review by ctverceles — COMMENTED** [review]
@mystbrent does this happen on your local? 
image

On kovan, im getting the same gas estimation err that i got on https://github.com/HaloDAO/pool-worker/pull/2#pullrequestreview-635220634 
image

**Comment by ctverceles** [comment]
As of this morning I still get the `revert ERC20: transfer to the zero address` error. 
image

on Kovan, 
image

I guess since that kovan err that was happening last week on https://github.com/HaloDAO/pool-worker/pull/2 is now resolved (kovan has moods) and this also has an err on local, then we can try checking whether or not this kovan err is related to the local err and not the kovan network itself @mystbrent

**Comment by mystbrent** [comment]
Local is working good here. There must be something causing the issue

image

**Comment by mystbrent** [comment]
Note: Kovan errors have been concluded to be a product of Kovan network issues.

**Comment by mystbrent** [comment]
Found cause of local error its because there was no setHaloChest call during the deploy instructions in deployAllLocal.

**Comment by mystbrent** [comment]
Updated to use halo-rewards submodule PR:
https://github.com/HaloDAO/halo-rewards/pull/14

Will test for approval

**Comment by ctverceles** [comment]
@mystbrent works on my local
Screen Shot 2021-04-20 at 1 17 31 PM
Screen Shot 2021-04-20 at 1 17 37 PM

and works on kovan! https://kovan.etherscan.io/address/0xe37b0E26d6f52ee36edF8a88Fbba7c532EC47Bd9

all good my end to merge!

**Comment by mystbrent** [comment]
Bumped halo-rewards submodule to https://github.com/HaloDAO/halo-rewards/commit/6e9b368de3e4cd4bbd4e76aab887360268d26433.

**Pull request merged**